SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUpstreamChecker.java

```diff
@@ -0,0 +1,157 @@
+  }
+
+  public void doSyncUpstreamForService(SingularityRequest singularityRequest) {
+    if (singularityRequest.isLoadBalanced() && noPendingDeploy()) {
+      final String singularityRequestId = singularityRequest.getId(); //TODO: lock on the requestId
+      final Optional<String> maybeDeployId = deployManager.getInUseDeployId(singularityRequestId);
+      if (maybeDeployId.isPresent()) {
+        final String deployId = maybeDeployId.get();
+        final Optional<SingularityDeploy> maybeDeploy = deployManager.getDeploy(singularityRequestId, deployId);
+        if (maybeDeploy.isPresent()) {
+          final SingularityDeploy deploy = maybeDeploy.get();
+          final Optional<String> loadBalancerUpstreamGroup = deploy.getLoadBalancerUpstreamGroup();
+          final SingularityLoadBalancerUpdate syncUpstreamsUpdate;
+          try {
+            syncUpstreamsUpdate = syncUpstreamsForService(singularityRequest, deploy, loadBalancerUpstreamGroup);
+            final LoadBalancerRequestId loadBalancerRequestId = syncUpstreamsUpdate.getLoadBalancerRequestId();
+            final BaragonRequestState syncUpstreamsState = lbClient.getState(loadBalancerRequestId).getLoadBalancerState();
+            switch (syncUpstreamsState) { //TODO: discuss what to do when the state is not SUCCESS, apart from logging
+              case SUCCESS:
+                LOG.info("Syncing upstreams for singularity request {} is successful. Load balancer request id is {}.", singularityRequestId, loadBalancerRequestId.toString());
+              case CANCELED:
+                LOG.info("Syncing upstreams for singularity request {} is canceled. Load balancer request id is {}.", singularityRequestId, loadBalancerRequestId.toString());
+              case FAILED:
+                LOG.info("Syncing upstreams for singularity request {} failed. Load balancer request id is {}.", singularityRequestId, loadBalancerRequestId.toString());
+              case INVALID_REQUEST_NOOP:
+                LOG.info("Syncing upstreams state for singularity request {} is invalid_request_noop. Load balancer request id is {}.", singularityRequestId, loadBalancerRequestId.toString());
+              case CANCELING:
+                LOG.info("Syncing upstreams for singularity request {} is canceling. Load balancer request id is {}.", singularityRequestId, loadBalancerRequestId.toString());
+              case UNKNOWN:
+                LOG.info("Syncing upstreams for singularity request {} is unknown. Load balancer request id is {}.", singularityRequestId, loadBalancerRequestId.toString());
+              case WAITING:
+                LOG.info("Syncing upstreams for singularity request {} is waiting. Load balancer request id is {}.", singularityRequestId, loadBalancerRequestId.toString());
+                break;
+            }
+          } catch (IOException | InterruptedException | ExecutionException | TimeoutException e) {
+            e.printStackTrace();
```

Comment:
Generally prefer using a `LOG.error("message", e)` here. This will send the exception through our normal logging stack and will also trigger a sentry automatically for anything at error level